packages/integrations/src/transactionactivity.ts

```diff
@@ -0,0 +1,152 @@
+import { EventProcessor, Hub, Integration, Scope, Span, SpanContext } from '@sentry/types';
+
+/** JSDoc */
+interface TransactionActivityOptions {
+  // onLocationChange: (info) => {
+  //   // info holds the location change api.
+  //   // if this returns `null` there is no transaction started.
+  //   return info.state.transaction || info.url;
+  // },
+  // onActivity?: (info) => {
+  //   return info.type !== 'xhr' || !info.url.match(/zendesk/);
+  // },
+  idleTimeout?: number;
+}
+
+/** JSDoc */
+interface Activity {
+  name: string;
+  span?: Span;
+}
+
+/** JSDoc */
+export class TransactionActivity implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = TransactionActivity.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'TransactionActivity';
+
+  /** JSDoc */
+  private static _options: TransactionActivityOptions;
+
+  /**
+   * Returns current hub.
+   */
+  private static _getCurrentHub?: () => Hub;
+
+  private static _activeTransaction?: Span;
+
+  private static _currentIndex: number = 0;
+
+  private static readonly _activities: { [key: number]: Activity } = {};
+
+  private static _debounce: number = 0;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options?: TransactionActivityOptions) {
+    TransactionActivity._options = {
```

Comment:
You override the class options here, should attach it to instance instead

Reply:
Yeah, this class only works as a "singleton". Having the options on the instance doesn't work since almost everything is on the static class.